ktor-client/ktor-client-features/ktor-client-json/common/src/io/ktor/client/features/json/JsonFeature.kt

```diff
@@ -79,13 +81,21 @@ class JsonFeature internal constructor(
             get() = _acceptContentTypes
 
         /**
-         * Adds accepted content types. Be aware that [ContentType.Application.Json] accepted by default is removed from
-         * the list if you use this function to provide accepted content types.
-         * It also affects `Accept` request header value.
+         * Adds accepted content types.
          */
         fun accept(vararg contentTypes: ContentType) {
             _acceptContentTypes += contentTypes
         }
+
+        /**
+         * Defines when to handle a given [ContentType] (used in addition to [accept]).
+         *
+         * By default accepts any `application/<...>+json` pattern.
+         */
+        var shouldHandle: (ContentType) -> Boolean = {
```

Comment:
It's easy to break source compatibility if we have `public` functional property.
Let's make it simple: instead of introducing the public field, let's make a config function:
```
fun filterContentType(block: (ContentType) -> Boolean {
 ...
}
```
And make the field at internal or private. I think the `filter` is the proper word here because it's naturally what we're doing.

Reply:
And we need at least one test using that API

Reply:
The advantage of a field is that it's composable. You can do:

```kotlin
httpClient.config {
    install(JsonFeature) {
        val prevFilter = filterContentType
        filterContentType = { prevFilter(it) || myOtherFilter(it) }
    }
}
```

With a function you lose that property unless you introduce a complicated API.

I do see your argument that a function can be extended with additional arguments in the future, but I believe that composability is more important in this case and also that a simple field is no different from inheritance/polymorphism where you override a single method. With inheritance you have the sample problem that changes to the function signature are breaking changes.

Reply:
That's a good use-case! So let's fit it in our usual ktor model: we can make the field public with `private` set, and introduce method as I mentioned before.

So we will get a nice builder-like API with the ability to reuse previously configured method!

Reply:
This is resolved with the ContentTypeMatcher solution.